runtime/Cpp/runtime/src/misc/IntervalSet.cpp

```diff
@@ -46,6 +46,21 @@ IntervalSet::IntervalSet(int n, ...) : IntervalSet() {
   }
 }
 
+IntervalSet::~IntervalSet()
+{
+}
+
+IntervalSet& IntervalSet::operator=(const IntervalSet& other)
+{
+  if (_readonly) {
+    throw IllegalStateException("can't alter read only IntervalSet");
+  }
```

Comment:
I'm not sure this is the intention of the original code. If you copy an interval set the old one goes completely, so it doesn't matter if it was read only or not, but instead takes the entire state from the source interval set (including it's read only value).